lib/internal/util/inspect.js

```diff
@@ -159,20 +159,31 @@ const isUndetectableObject = (v) => typeof v === 'undefined' && v !== undefined;
 
 // These options must stay in sync with `getUserOptions`. So if any option will
 // be added or removed, `getUserOptions` must also be updated accordingly.
-const inspectDefaultOptions = ObjectSeal({
+const inspectDefaultOptions = {
   showHidden: false,
   depth: 2,
   colors: false,
   customInspect: true,
   showProxy: false,
-  maxArrayLength: 100,
+  maxItemLength: 100,
   maxStringLength: 10000,
   breakLength: 80,
   compact: 3,
   sorted: false,
   getters: false,
   numericSeparator: false,
+};
+ObjectDefineProperty(inspectDefaultOptions, 'maxArrayLength', {
+  __proto__: null,
+  get() {
+    return inspectDefaultOptions.maxItemLength;
```

Comment:
I think the *logic* needs to always check both fields, instead of trying to "fix" it in the defaults. That way it also handles when *users* pass both options.

Reply:
What about this logic:
- No default value
- If `options.maxItemLength` is a number, use it
- Otherwise, use `options.maxArrayLength`

Reply:
Using both options is also tricky as they might contain different values and that could be considered a bug.

Maybe it's best to just stick to the current name and outline that the name is historical in the documentation?

Reply:
We could also throw if you pass both?

(however I can still see a use case for having a different max length for arrays as for maps/sets)

Reply:
Thank you all! I think `maxItemLength` is more reasonable name and easier to understand the use, but at the same time, using both options as the same usage is tricky and can lead to a bug.

In my opinion, the first step would be to add `maxArrayLength` support to `Set` and `Map` to provide users with a way to truncate a long iterable entries. Then, after discussion and consensus with user opinions on GitHub issues, we will create a follow up PR if needed.
So I would like to improve `maxArrayLength` support in this PR. What do you think?

Reply:
I still think it’d be better to have two options, per https://github.com/nodejs/node/pull/43576#discussion_r910141616.

If passing both options (which has a use case) is not desired, we can make it throw in that event.

Reply:
I think sticking with one option (despite its bad name) is fine for this PR. Having a separate option for non-array iterables sounds fine to me but it can be discussed in its own PR.

Reply:
I agree with @aduh95.

Reply:
This wouldn't be something that could be changed after release, though, without a semver-major - that's why I think it needs to be addressed in this PR.

Reply:
`WeakMap`s and `WeakSet`s are already supported by `maxArrayLength`, if we create a new option with a broader name it would need to apply to those anyway. I don't think it would necessarily require a semver-major change if we ensure to keep backward compatible fallbacks.

Reply:
It's a fair point that the option is already overloaded; it's just that I would want to be able to control the max length of each type individually, and this change makes that harder and less likely.

Reply:
I mean, if you or someone else were to open an alternative PR with the design you prefer, that could be discussed, but as is, this PR is ready, and does improve the current API; we can't force anyone to work on it, it wouldn't be fair to block this PR if no one is going to work on the alternative API design anyway.

Reply:
I would be happy to work on it; the blocker is fear of potential objections by collabs, and fear of biasing existing PRs.

Can this PR wait while i work on the changes? I can PR them into @cola119’s fork so this PR can be updated.

Reply:
> the blocker is fear of potential objections by collabs

Hopefully the fear of a not-as-ideal design landing is greater than the fear of being blocked – and FWIW, I don't see a single reason for your PR to be blocked.



> Can this PR wait while i work on the changes?

If you get an open (or draft) PR before we got a green CI on this PR (which given the current state of the CI is very achievable), that's a given. Otherwise, we can land this PR and add https://github.com/nodejs/node/labels/backport-blocked-v18%2Ex label to give you more time.

Reply:
Either would be appreciated. I'll work on it over the next few days.

Reply:
I welcome to wait your implementation. Thank you.